aten/src/ATen/native/native_functions.yaml

```diff
@@ -6437,6 +6440,16 @@
     SparseCsrCUDA: sparse_sampled_addmm_sparse_csr_cuda
     SparseCsrCPU: sparse_sampled_addmm_sparse_csr_cpu
 
+- func: _sparse_mm_reduce_impl(Tensor self, Tensor other, str reduce, *, Tensor? row_indices=None, Tensor? ccol_indices=None, Tensor? csr2csc=None) -> (Tensor, Tensor)
+  python_module: sparse
+  dispatch:
+    SparseCsrCPU: _sparse_mm_reduce_impl_sparse_csr_cpu
+
+- func: _sparse_mm_reduce_impl_backward(Tensor self, Tensor grad_out, Tensor weight, str reduce, Tensor arg_out, Tensor row_indices, Tensor ccol_indices, Tensor csr2csc, bool[2] output_mask) -> (Tensor, Tensor)
```

Comment:
I removed all the meta tensors for backward in the backward function signature, which would make the upper level API more clean.

The computation of these meta tensors are local to backward right now.

**TODO:** optimize the meta tensor calculation, right now it would be rather slow since the `argsort` is sequential. I will have it optimized, but probably going to take some effort, not very easy to do.